.github/workflows/deploy_to_aws.yml

```diff
@@ -412,3 +362,83 @@ jobs:
+    steps:
+      - name: Get Environment Name for ${{ vars.ENV_NAME }}
+        id: get_env_name
+        uses: Entepotenz/change-string-case-action-min-dependencies@v1
+        with:
+          string: ${{ vars.ENV_NAME }}
+      - name: Checkout repo
+        uses: actions/checkout@v4
+      - name: Get OpenSearch domain, filesystem and access point IDs for ${{ vars.ENV_NAME }}
+        if: ${{  inputs.clear-opensearch == true }}
+        id: export_variables
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          AWS_DEFAULT_REGION: "eu-west-1"
+        run: |
+          OS_DOMAIN_NAME=$(echo "${{ vars.ENV_NAME }}-os-domain" | tr '[:upper:]' '[:lower:]')
+          OPENSEARCH_DOMAIN=$(aws \
+                       es describe-elasticsearch-domains --domain-names $OS_DOMAIN_NAME \
+                       --query "DomainStatusList[].Endpoints.vpc" --output text)
+          EFS_ID=$(aws \
+                       efs describe-file-systems \
+                       --query "FileSystems[?Tags[?Key=='Environment' && Value=='${{ vars.ENV_NAME }}']].FileSystemId" \
+                       --output text)
+          EFS_AP_ID=$(aws \
+                          efs describe-access-points \
+                          --query "AccessPoints[?FileSystemId=='$EFS_ID'].AccessPointId" \
+                          --output text)
+          echo "EFS_ID=$EFS_ID" >> $GITHUB_OUTPUT
+          echo "EFS_AP_ID=$EFS_AP_ID" >> $GITHUB_OUTPUT
+          echo "OPENSEARCH_DOMAIN=$OPENSEARCH_DOMAIN" >> $GITHUB_OUTPUT
+      - name: Clear the OpenSearch indexes for ${{ vars.ENV_NAME }}
+        if: ${{  inputs.clear-opensearch == true }}
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          OPENSEARCH_DOMAIN: ${{ steps.export_variables.outputs.OPENSEARCH_DOMAIN }}
+          EFS_AP_ID: ${{ steps.export_variables.outputs.EFS_AP_ID }}
+          EFS_ID: ${{ steps.export_variables.outputs.EFS_ID }}
+          BASTION_IP: ${{ vars.BASTION_IP }}
+        run: |
+          cd ./deployment/terraform/clear_opensearch
+          mkdir -p script
+          mkdir -p ssh
+          echo -e "Host *\n\tStrictHostKeyChecking no\n\n" > ssh/config
+          printf "%s\n" "${{ secrets.SSH_PRIVATE_KEY }}" > ssh/id_rsa
+          echo "" >> ssh/id_rsa
+          echo -n ${{ vars.BASTION_IP }} > script/.env
+          envsubst < clear_opensearch.sh.tpl > script/clear_opensearch.sh
+          envsubst < run.sh.tpl > script/run.sh
+          docker run --rm  \
+              -v ./script:/script \
+              -v ./ssh:/root/.ssh \
+              kroniak/ssh-client bash /script/run.sh
+
+  start_logstash:
```

Comment:
Please, add a check to ensure it only starts if restore-db or clear-opensearch are true. Without this, we will start Logstash even when we don't need to clear OpenSearch.

Reply:
We discussed that at the post-deployment stage there may be migrations affecting the data schema and therefore Logstash should not be running at this time. Will there be a situation where the database does not need to be restored, but migrations will still occur?

Reply:
@roninzp 
Why can't this check be done on the job level? To avoid starting this job. Will it block the next jobs if the check is at the job level?

Reply:
if we add this check on job level, all followed jobs be skipped every time when triger not be set to true

Reply:
Got it, thank you!